airflow/www_rbac/views.py

```diff
@@ -1140,6 +1140,39 @@ def dagrun_cancel(self):
         return self._clear_dag_tis(dag, None, None, origin,
                                    recursive=True, confirmed=confirmed, cancel=True)
 
+    @expose('/cancel_dagrun', methods=['POST'])
+    @has_dag_access(can_dag_edit=True)
+    @has_access
+    @action_logging
+    def cancel_specific_dagrun(self):
+        # cancel specific dagrun
+        dag_id = request.form['dag_id']
+        origin = request.form['origin']
+        # it seems sqlalchemy wont work with Pendulum datetimes so we have to get its private datetime
+        execution_date = pendulum.parse(request.form['execution_date'])._datetime
+        confirmed = request.form.get('confirmed') == "true"
+        dag = dagbag.get_dag(dag_id)
+        return self._clear_dag_tis(dag, execution_date, execution_date, origin,
+                                   recursive=True, confirmed=confirmed, cancel=True)
+
+    @expose('/cancel_task_instance', methods=['POST'])
+    @has_dag_access(can_dag_edit=True)
+    @has_access
+    @action_logging
+    def cancel_task_instance(self):
```

Comment:
can we name this `cancel` or change `clear` to be `clear_task_instance` for consistency? the name of the function is also what gets used as the event name in the Log table so keeping it consistent would be nice. changing `clear` to `clear_task_instance` might be preferable so we have more granularity on the specific events